scripts/us_census/pep/monthly_population_estimate/preprocess.py

```diff
@@ -378,30 +403,266 @@ def process(self):
+
+    Returns:
+        res (Series) : Concatenated DataFrame Columns
+    """
+    res = col[0]
+    if col[1] is None:
+        return res
+    res = col[0] + ' ' + col[1]
+    return res
+
+
+def download_files():
+    """
+    This method allows to download the input files.
+    """
+    global _FILES_TO_DOWNLOAD
+    session = requests.session()
+    max_retry = 5
+    for file_to_dowload in _FILES_TO_DOWNLOAD:
+        file_name = None
+        url = file_to_dowload['download_path']
+        if 'file_name' in file_to_dowload and len(
+                file_to_dowload['file_name'] > 5):
+            file_name = file_to_dowload['file_name']
+        else:
+            file_name = url.split('/')[-1]
+        retry_number = 0
+
+        is_file_downloaded = False
+        while is_file_downloaded == False:
+            try:
+                df = None
+                file_name = url.split("/")[-1]
+                if ".xls" in url:
+                    df = pd.read_excel(url, header=_HEADER)
+                    df.to_excel(os.path.join(_INPUT_FILE_PATH, file_name),
+                                index=False,
+                                header=False,
+                                engine='xlsxwriter')
+                elif ".csv" in url:
+                    file_name = file_name.replace(".csv", ".xlsx")
+                    df = pd.read_csv(url, header=None)
+                    df = _clean_csv_file(df)
+                    df.to_excel(os.path.join(_INPUT_FILE_PATH, file_name),
+                                index=False,
+                                engine='xlsxwriter')
+                elif ".txt" in url:
+                    file_name = file_name.replace(".txt", ".xlsx")
+                    cols = [
+                        "Year and Month", "Date", "Resident Population",
+                        "Resident Population Plus Armed Forces Overseas",
+                        "Civilian Population",
+                        "Civilian NonInstitutionalized Population"
+                    ]
+                    df = pd.read_table(url,
+                                       index_col=False,
+                                       delim_whitespace=True,
+                                       engine='python',
+                                       skiprows=17,
+                                       names=cols)
+                    # Skipping 17 rows as the initial 17 rows contains the information about
+                    # the file being used, heading files spread accross multiple lines and
+                    # other irrelevant information like source/contact details.
+                    df = _clean_txt_file(df)
+                    # Multiplying the data with scaling factor 1000.
+                    for col in df.columns:
+                        if "year" not in col.lower():
+                            df[col] = df[col].apply(_mulitply_scaling_factor)

```

Comment:
will this multiply 1000 other columns like FIPS code?

Can we move this logic to the processing function so `input_files/` has the exact copy as downloaded from source?

Reply:
A .   I verified processed data with raw df its multiply only for reqiured column.
B.    we are storing the raw data before processing it.